sklearn/cluster/tests/test_affinity_propagation.py

```diff
@@ -137,8 +137,10 @@ def test_affinity_propagation_predict_non_convergence():
 
     # At prediction time, consider new samples as noise since there are no
     # clusters
+    to_predict = np.array([[2, 2], [3, 3], [4, 4]])
+    assert_warns(ConvergenceWarning, af.predict, to_predict)
```

Comment:
please just put `y = ` before this line rather than calling predict (and triggering the warning) again

Reply:
I changed that and also added an `assert_warns` around the call to `fit` earlier in the same test that (intentionally) triggers a `ConvergenceWarning` as well (although that one was already there).